internal/pkg/service/stream/storage/node/readernode/sliceupload/operator.go

```diff
@@ -167,34 +198,40 @@ func (o *operator) checkSlices(ctx context.Context, wg *sync.WaitGroup) {
 	})
 }
 
-func (o *operator) checkSlice(ctx context.Context, data *sliceData) {
+func (o *operator) checkSlice(ctx context.Context, slice *sliceData) {
 	// Log/trace slice details
-	ctx = ctxattr.ContextWith(ctx, data.Attrs...)
+	ctx = ctxattr.ContextWith(ctx, slice.Attrs...)
 
 	// Prevent multiple checks of the same slice
-	if !data.Lock.TryLock() {
+	if !slice.Lock.TryLock() {
 		return
 	}
-	defer data.Lock.Unlock()
+	defer slice.Lock.Unlock()
 
 	// Slice has been modified by some previous check, but we haven't received an updated version from etcd yet
-	if data.Processed {
+	if slice.Processed {
+		return
+	}
+
+	if !slice.Retryable.Allowed(o.clock.Now()) {
 		return
 	}
 
-	if !data.Retryable.Allowed(o.clock.Now()) {
+	// Skip file import if sink is deleted or disabled
+	sink, ok := o.sinks.Get(slice.SinkKey)
+	if !ok || !sink.Enabled {
```

Comment:
I would move it after `volume` check. It does not need `repository` and etcd to be checked when `slice` has no volume to be read on.

Reply:
It is loaded from the mirror, so it is fast.